webapp/src/ts/modals/training-cards/training-cards.component.ts

```diff
@@ -0,0 +1,217 @@
+import { Component, NgZone, OnDestroy, OnInit } from '@angular/core';
+import { BsModalRef } from 'ngx-bootstrap/modal';
+import { Store } from '@ngrx/store';
+import { combineLatest, Subscription } from 'rxjs';
+
+import { MmModalAbstract } from '@mm-modals/mm-modal/mm-modal';
+import { XmlFormsService } from '@mm-services/xml-forms.service';
+import { EnketoService } from '@mm-services/enketo.service';
+import { Selectors } from '@mm-selectors/index';
+import { GlobalActions } from '@mm-actions/global';
+import { GeolocationService } from '@mm-services/geolocation.service';
+import { TranslateService } from '@mm-services/translate.service';
+import { TelemetryService } from '@mm-services/telemetry.service';
+
+@Component({
+  selector: 'training-cards-modal',
+  templateUrl: './training-cards.component.html'
+})
+export class TrainingCardsComponent extends MmModalAbstract implements OnInit, OnDestroy {
+
+  constructor(
+    bsModalRef: BsModalRef,
+    private ngZone: NgZone,
+    private store: Store,
+    private xmlFormsService: XmlFormsService,
+    private enketoService: EnketoService,
+    private geolocationService: GeolocationService,
+    private translateService: TranslateService,
+    private telemetryService: TelemetryService,
+  ) {
+    super(bsModalRef);
+    this.globalActions = new GlobalActions(this.store);
+  }
+
+  static id = 'training-cards-modal';
+  private geoHandle:any;
+  private globalActions;
+  form;
+  trainingForm;
+  formWrapperId = 'training-cards-form';
+  modalTitleKey = 'training_cards.modal.title';
+  loadingContent;
+  contentError;
+  hideModalFooter;
+  errorTranslationKey
+  enketoError;
+  enketoStatus;
+  enketoSaving;
+  telemetryData;
+  showConfirmExit;
+  subscription: Subscription = new Subscription();
+
+  ngOnInit() {
+    this.telemetryData = { preRender: Date.now() };
+    this.reset();
+    this.subscribeToStore();
+  }
+
+  ngOnDestroy() {
+    this.subscription.unsubscribe();
+    this.geoHandle && this.geoHandle.cancel();
+    // old code checked whether the component is reused before unloading the form
+    // this is because AngularJS created the new "controller" before destroying the old one
+    // in Angular, unless specific RouteReuseStrategies are employed, components are always
+    // destroyed before new ones are created
+    // see https://github.com/angular/angular/blob/10.2.x/packages/router/src/operators/activate_routes.ts#L37
+    // for Angular behavior
+    // see https://github.com/medic/cht-core/issues/2198#issuecomment-210202785 for AngularJS behavior
+    this.enketoService.unload(this.form);
+    this.globalActions.clearEnketoStatus();
+  }
+
+  private loadForm() {
+    return this.ngZone.runOutsideAngular(() => this._loadForm());
+  }
+
+  private async _loadForm() {
```

Comment:
The `_` prefix was used in the old days when we didn't have `private` members. Now it's not needed. Rename this to `loadForm`

Reply:
I asked the same when we did the angular migration, but at the end it turned out as a way to avoid name conflicts (still a thing in some languages/practices), between the function that contains ngZone ([here](https://github.com/medic/cht-core/pull/7628/files/4ac0f6a2a0f6224c57539aeeacf072f79481a41d#diff-94d07e089bc01aba387ccbb407e1b7ed08b021c77672da72e2511b1a92b4cef9R73)) and the function that runs outsize "angular ng zone". At the end we're doing this here to keep consistency with everywhere else in the code.

Reply:
For now I've left the prefix to keep consistency in the code